File: web/src/models/BlockDataModel.ts
/**
 * Stores data used to process a block.
 */
interface BlockDataModel {
  /**
   * Material ID ordinal.
   */
  material: number;

  /**
   * Additional block data.
   */
  data?: string;

  /**
   * Sky light value.
   */
  skyLight?: number;

  /**
   * Emitted light value.
   */
  emittedLight?: number;

  /**
   * Biome containing the block.
   */
  biome?: number;

  /**
   * Temperature at the block.
   */
  temperature?: number;

  /**
   * Humidity at the block.
   */
  humidity?: number;
}

export default BlockDataModel;

File: plugin/src/main/java/com/jonathanpotts/blockmaps/models/BlockDataModel.java
package com.jonathanpotts.blockmaps.models;

/**
 * Stores data used to process a block.
 */
public class BlockDataModel {
  /**
   * Material ID ordinal.
   */
  public Integer material;

  /**
   * Additional block data.
   */
  public String data;

  /**
   * Sky light value.
   */
  public Integer skyLight;

  /**
   * Emitted light value.
   */
  public Integer emittedLight;

  /**
   * Biome containing the block.
   */
  public Integer biome;

  /**
   * Temperature at the block.
   */
  public Double temperature;

  /**
   * Humidity at the block.
   */
  public Double humidity;
}
